controller/workitemtype.go

```diff
@@ -135,9 +135,9 @@ func (c *WorkitemtypeController) List(ctx *app.ListWorkitemtypeContext) error {
 func (c *WorkitemtypeController) ListSourceLinkTypes(ctx *app.ListSourceLinkTypesWorkitemtypeContext) error {
 	return application.Transactional(c.db, func(appl application.Application) error {
 		// Test that work item type exists
-		_, err := appl.WorkItemTypes().Load(ctx.Context, ctx.SpaceID, ctx.WitID)
-		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, err)
+		exists, err := appl.WorkItemTypes().Exists(ctx.Context, ctx.WitID.String())
+		if !exists || err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
```

Comment:
Also, here's a potential implicit cast from `InternalError` to a `NotFound` error. I think it is enough to just do `return jsonapi.JSONErrorResponse(ctx, err)`. Our JSONAPI error handle will take care in translating that underlying error into a proper JSONAPI error with status codes and so forth because it knows about our error codes.